posts/2023-09-29-open-liberty-on-aws-ecs-fargate.adoc

```diff
@@ -0,0 +1,684 @@
+
+If you don't have experience with Open Liberty and MicroProfile or wish to understand more about the application used in this blog follow the steps in <<Create your MicroProfile application>> and <<Uploading the container to a registry>>.
+
+=== Create your MicroProfile application
+In this post, we use https://openliberty.io/guides/getting-started.html[Open Liberty Getting Started guide] to get a simple MicroProfile application that is well suited to run in a serverless environment.
+
+Follow the guide up to and including https://openliberty.io/guides/getting-started.html#running-the-application-in-a-docker-container[Running the application in a Docker container]
+
+=== Uploading the container to a registry
+If you wish to use the image created in <<Create your MicroProfile application>> then it needs to be uploaded to a internet accessible container registry.
+
+Amazon ECS supports a range of container registries outlined in their https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_image[Image Registry requirements].
+
+Depending on your choice of registry follow the steps provided by it for uploading your image.
+
+As this blog requires an AWS account, you are most likely to have access to Amazon ECR to act as your registry. To upload the image use the https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html[Get Started] instructions for Amazon ECR.
+
+=== Creating your Amazon ECS Cluster
+To create your Amazon ECS cluster follow step 1. in https://docs.aws.amazon.com/AmazonECS/latest/developerguide/getting-started-fargate.html[Getting started with the console using Linux containers on AWS Fargate].
+
+=== Creating your Task Definition
+Amazon ECS runs either Services or Jobs that use https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html[Task Definitions] to define the runtime configuration for the task. The task definition includes some of the following properties:
+
+* Container Image URL
+* CPU & Memory
+* Port Mappings
+* Environment variables
+* Compatibilities
+
+The values of these properties cannot be overridden by the service or job definition that execute the task definition. For example, if the same task definition was used in Development and Production deployments, then they would share these properties. As such it is recommended that separate task definitions are defined per environment.
+
+The instructions below will use a publicly accessible Open Liberty getting started guide container image. If you are using your own container, substitute `icr.io/appcafe/open-liberty/samples/getting-started` with the path for your image.
+
+There are two methods to create your task definition, via a UI wizard or via applying a JSON definition.
```

Comment:
Don't give both examples in this blog. Pick one and stick with it for the example. Fine to mention that they can do both, but it's too much to have both in the guide. 
We don't want this blog to be documenting the system. We just want it to be more of a quick tutorial. Assume a golden path and don’t give options or alternatives. Link out to docs for more info.
Pick the method (UI vs JSON) based on what developers are most likely to want to use. If it’s a toss-up about preference, then pick the one that’s easiest to use if new to it. If they’re both easy, then pick the one that’s easiest to document.